### PR TITLE
Enhancements to the launch files

### PR DESCRIPTION
**Pull request by melihokur17**
Removed the ground plane  from the pool world and brought the pool to z=-2, making the water surface equal to z=0.

Added new arguments to the launch files to allow the desired world to be launched with the command.

`roslaunch auv_sim_bringup start.launch namespace:=taluy use_gui:=1 world:=sea 
`

while this command starts gazebo with sea, removing world arg on the command starts the pool by default.

### CODE REVIEW

**Comment by senceryazici** [comment]
Again this looks also fine, I see what you're achieving. I think the context of this PR is close to #90. 
Just left some comments for #90 so maybe you and @eminmeydanoglu can review my comments there and both of your approaches on these PR's, and come up with something hopefully "more singular" :D

File: auv_sim/auv_sim_bringup/launch/inc/start_world.launch.xml [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
   <arg name="debug" default="false" />
-  <arg name="world_name" default="$(find auv_sim_description)/worlds/seabed.world" />
   <arg name="namespace" default="taluy" />
-  <arg name="velocity_control" default="true" />
-  <arg name="joy_id" default="0" />
+  <arg name="world" default="pool" />
+  <arg name="gui" default="true" />
+
+  <arg name="world_name" value="$(eval find('auv_sim_description') + '/worlds/' + arg('world') + '.world')" />
+
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/inc/start_world.launch.xml:9]: Dynamic world selection, using the world argument. roslaunch auv_sim_bringup start.launch world:=seabed

File: auv_sim/auv_sim_bringup/launch/inc/start_world.launch.xml [hunk 2]
```diff
@@ -16,8 +17,9 @@
     <arg name="verbose" value="true" />
   </include>
 
-  <node name="publish_world_models" pkg="uuv_assistants" type="publish_world_models.py"
+  <node if="$(eval arg('world') == 'seabed')" name="publish_world_models" pkg="uuv_assistants" type="publish_world_models.py"
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/inc/start_world.launch.xml:20]: Only run `publish_world_models` if `world `is `seabed`.

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 3]
```diff
@@ -1,30 +1,30 @@
 <?xml version="1.0"?>
 <launch>
+  <env name="GAZEBO_MODEL_PATH" value="$(find auv_sim_description)/models:${GAZEBO_MODEL_PATH}"/>
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:3]: the automated gazebo model path selector

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 4]
```diff
@@ -1,30 +1,30 @@
...
+
 
-  <arg name="x" value="9.0" />
-  <arg name="y" value="-3.0" />
-  <arg name="z" value="-87.0" />
+  <!--Initial robot poisition and orientation-->
+  <arg name="x" value="0.0" />
+  <arg name="y" value="0.0" />
+  <arg name="z" value="-1.50" />
+  <arg name="z_sea" value="-87" if="$(eval arg('world') == 'seabed')" />
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:17]: We want z to be -87 when using seabed, and -2 when using the pool.

File: auv_sim/auv_sim_bringup/launch/inc/start_pool_world.launch.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  <include file="$(find gazebo_ros)/launch/empty_world.launch">
+    <arg name="world_name" value="$(arg world_name)" />
+    <arg name="use_sim_time" value="true" />
+    <arg name="gui" value="$(arg gui)" />
+    <arg name="headless" value="false" />
+    <arg name="debug" value="$(arg debug)" />
+    <arg name="verbose" value="true" />
+  </include>
+
+</launch>
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/inc/start_pool_world.launch.xml:17]: Separated the launch files for each world to make it easy and clean to add world-specific nodes

File: auv_sim/auv_sim_bringup/launch/inc/start_pool_world.launch.xml [hunk 6]
```diff
@@ -16,8 +14,4 @@
     <arg name="verbose" value="true" />
   </include>
 
-  <node name="publish_world_models" pkg="uuv_assistants" type="publish_world_models.py"
-    output="screen">
-    <rosparam command="load" file="$(find auv_sim_bringup)/config/world_models.yaml" />
-  </node>
-</launch>
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/inc/start_pool_world.launch.xml:23]: Separated the launch files for each world to make it easier to add world-specific nodes

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 7]
```diff
@@ -1,30 +1,29 @@
 <?xml version="1.0"?>
 <launch>
+  <env name="GAZEBO_MODEL_PATH" value="$(find auv_sim_description)/models:${GAZEBO_MODEL_PATH}"/>
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:3]: Check https://github.com/itu-auv/auv-software/pull/90/files#r1876316169

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 8]
```diff
@@ -1,30 +1,29 @@
...
+
 
-  <arg name="x" value="9.0" />
-  <arg name="y" value="-3.0" />
-  <arg name="z" value="-87.0" />
+  <!--Initial robot poisition and orientation-->
+  <arg name="x" value="0.0" />
+  <arg name="y" value="0.0" />
+  <arg name="z" value="-1.50" />
+  <arg name="z_sea" value="-87" if="$(eval arg('world') == 'seabed')" />
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:17]: We want the z value to be -87 if using the seabed world

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 9]
```diff
@@ -1,30 +1,29 @@
...
 
-  <include file="$(find auv_sim_bringup)/launch/inc/start_world.launch.xml">
-    <arg name="gui" default="$(arg use_gui)" />
-    <arg name="debug" default="$(arg debug)" />
-    <arg name="world_name" default="$(arg world_name)" />
-    <arg name="namespace" default="$(arg namespace)" />
-    <arg name="velocity_control" default="$(arg velocity_control)" />
-    <arg name="joy_id" default="$(arg joy_id)" />
+
+  <include file="$(find auv_sim_bringup)/launch/inc/start_$(arg world)_world.launch.xml">
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:23]: I believe this is an elegant way to select worlds:  
- Naming format for the world launch files: `start_WORLD_launch.xml`  
- Insert the world argument, and done.

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 10]
```diff
@@ -35,7 +34,7 @@
     <arg name="reference_frame" value="$(arg reference_frame)" />
     <arg name="x" value="$(arg x)" />
     <arg name="y" value="$(arg y)" />
-    <arg name="z" value="$(arg z)" />
+    <arg name="z" value="$(eval arg('z_sea') if arg('world') == 'seabed' else arg('z'))" />
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:37]: Use the z value specified for seabed if world = seabed

**Comment by eminmeydanoglu** [comment]
@senceryazici waiting for you review

File: auv_sim/auv_sim_bringup/launch/start.launch [hunk 11]
```diff
@@ -1,49 +1,54 @@
...
   <arg name="use_gui" default="true" />
   <arg name="debug" default="false" />
-  <arg name="world_name" default="$(find auv_sim_description)/worlds/seabed.world" />
-  <arg name="velocity_control" default="true" />
-  <arg name="joy_id" default="0" />
+  <arg name="world" default="pool" />
+  <arg name="use_ned_frame" default="false" />
+
+    <!--Run rviz-->
+  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find taluy_description)/rviz/yolo_cam.rviz"
```
Comment (eminmeydanoglu) [inline auv_sim/auv_sim_bringup/launch/start.launch:13]: Added rviz launch. Also added a new rviz config `yolo_cam.rviz` to display yolo camera output. Convenient for simulation.

**Comment by eminmeydanoglu** [comment]
Hey @senceryazici, you are welcome to review—everything is all set on our end :D

File: auv_sim/auv_sim_bringup/launch/inc/start_bridge.launch [hunk 12]
```diff
@@ -18,17 +18,21 @@
     </node>
   </group>
 
-  <!-- <include file="$(find auv_control)/launch/start.launch">
+  
+  <!--
+  <include file="$(find auv_control)/launch/start.launch">
```
Comment (senceryazici) [inline auv_sim/auv_sim_bringup/launch/inc/start_bridge.launch:23]: This is now moved to `start_navigation.launch` right? So delete the commented lines?

File: auv_sim/auv_sim_bringup/launch/inc/start_seabed_world.launch [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+        <arg name="use_sim_time" value="true" />
+        <arg name="gui" value="$(arg gui)" />
+        <arg name="headless" value="false" />
+        <arg name="debug" value="$(arg debug)" />
+        <arg name="verbose" value="true" />
+    </include>
+
+    <node name="publish_world_models" pkg="uuv_assistants" type="publish_world_models.py"
+    output="screen">
+    <rosparam command="load" file="$(find auv_sim_bringup)/config/world_models.yaml" />
```
Comment (senceryazici) [inline auv_sim/auv_sim_bringup/launch/inc/start_seabed_world.launch:19]: Format this file please.

File: auv_sim/auv_sim_bringup/launch/start_gazebo.launch [hunk 14]
```diff
@@ -1,49 +1,55 @@
...
 
-  <arg name="x" value="9.0" />
-  <arg name="y" value="-3.0" />
-  <arg name="z" value="-87.0" />
+
+  <!--Initial robot poisition and orientation-->
+  <arg name="x" value="0.0" />
+  <arg name="y" value="0.0" />
+  <arg name="z" value="-1.50" />
+  <arg name="z_sea" value="-87" if="$(eval arg('world') == 'seabed')" />
```
Comment (senceryazici) [inline auv_sim/auv_sim_bringup/launch/start_gazebo.launch:22]: I think this is rather a complicated logic here. Maybe use:

```xml

  <group if="$(eval arg('world') == 'seabed')">
    <arg name="x" value="0.0" />
    <arg name="y" value="0.0" />
    <arg name="z" value="-87.0" />
  </group>


 <group if="$(eval arg('world') == 'pool')">
    <arg name="x" value="0.0" />
    <arg name="y" value="0.0" />
    <arg name="z" value="-1.5" />
  </group>
```

Though I'm not sure if this is the case for launch arguments.

File: auv_sim/auv_sim_bringup/launch/start_gazebo.launch [hunk 15]
```diff
@@ -1,49 +1,55 @@
...
+  <include file="$(find auv_sim_bringup)/launch/inc/spawn_robot.launch">
     <arg name="debug" value="$(arg debug)" />
     <arg name="namespace" value="$(arg namespace)" />
     <arg name="control_rate" value="$(arg control_rate)" />
     <arg name="use_ned_frame" value="$(arg use_ned_frame)" />
     <arg name="reference_frame" value="$(arg reference_frame)" />
     <arg name="x" value="$(arg x)" />
     <arg name="y" value="$(arg y)" />
-    <arg name="z" value="$(arg z)" />
+    <arg name="z" value="$(eval arg('z_sea') if arg('world') == 'seabed' else arg('z'))" />
```
Comment (senceryazici) [inline auv_sim/auv_sim_bringup/launch/start_gazebo.launch:42]: above recommendation will eliminate this complex logic too.

File: auv_sim/auv_sim_bringup/launch/start_gazebo.launch [hunk 16]
```diff
@@ -1,49 +1,55 @@
...
   </include>
 
-  <include file="$(find auv_sim_bringup)/launch/inc/start_bridge.launch.xml">
+  <include file="$(find auv_sim_bringup)/launch/inc/start_bridge.launch">
     <arg name="namespace" value="$(arg namespace)" />
     <arg name="control_rate" value="$(arg control_rate)" />
-    <arg name="use_gui" default="$(arg use_gui)" />
+    <arg name="use_gui" value="$(arg use_gui)" />
   </include>
+  <!--start camera process-->
```
Comment (senceryazici) [inline auv_sim/auv_sim_bringup/launch/start_gazebo.launch:53]: If this is a todo, then add

```xml
<!-- TODO (someusername): add start camera process here -->
```

**Review by senceryazici — CHANGES_REQUESTED** [review]
Good work, requested small changes. Other than that this is good to go.

File: auv_sim/auv_sim_bringup/launch/inc/start_pool_world.launch [hunk 17]
```diff
@@ -1,11 +1,10 @@
 <?xml version="1.0"?>
 <launch>
-  <arg name="gui" default="true" />
+  <env name="GAZEBO_MODEL_PATH" value="$(find auv_sim_description)/models:${GAZEBO_MODEL_PATH}"/>
```
[the remark this answers is not in the document]
Comment (eminmeydanoglu) [reply auv_sim/auv_sim_bringup/launch/inc/start_pool_world.launch:3]: Have you take a look at https://github.com/itu-auv/auv-software/pull/90#discussion_r1876316169 ?

File: auv_sim/auv_sim_bringup/launch/start_gazebo.launch (hunk 14 above)
Comment (eminmeydanoglu) [reply auv_sim/auv_sim_bringup/launch/start_gazebo.launch:22]: When using a group, its arguments apply only within that group. So your recommendation causes the error: RLException: arg 'x' is not defined.

One could first define the arguments and then try to override in groups:
```xml
<!-- Define default arguments outside groups -->
<arg name="x" default="0.0" />
<arg name="y" default="0.0" />
<arg name="z" default="-1.5" />

<!-- Override defaults based on world -->
<group if="$(eval arg('world') == 'seabed')">
    <arg name="x" value="$(arg x)" />
    <arg name="y" value="$(arg y)" />
    <arg name="z" value="-87.0" />
</group>

<group if="$(eval arg('world') == 'pool')">
    <arg name="x" value="$(arg x)" />
    <arg name="y" value="$(arg y)" />
    <arg name="z" value="-1.5" />
</group>
```
But now this gives an override error:
RLException: Invalid <arg> tag: arg 'x' has already been declared. 

So there doesnt seem to be an easy way to group like your suggestion. However, I think this new approach is good and flexible enough (at least for now):
```xml
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="-1.5" if="$(eval arg('world') == 'pool')" />
    <arg name="z" default="-87.0" if="$(eval arg('world') == 'seabed')" />
    <arg name="roll" value="0" />
    <arg name="pitch" value="0" />
    <arg name="yaw" value="-1.8" />
```
and then simply
```xml
<arg name="z" value="$(arg z)" />
```

**Review by senceryazici — APPROVED** [review]
Good work

**Pull request merged**